src/main.cpp

```diff
@@ -1,19 +1,234 @@
 #include "main.hpp"
 
+using std::placeholders::_1;
+
+#define ENUM_TO_STR(member) \
+    case member:            \
+        return #member
+
+WaypointNode::WaypointNode() : CommonNode("waypoint_node") {
+    // Create a subscription for the "control" topic
+    control_subscription = this->create_subscription<interfaces::msg::Control>(
+        common_lib::topic_names::Control, 10,
+        std::bind(&WaypointNode::callback_control, this, _1));
+
+    // Create a subscription for the "uav_gps_position" topic
+    gps_position_subscription =
+        this->create_subscription<interfaces::msg::GPSPosition>(
+            common_lib::topic_names::GPSPosition, 10,
+            std::bind(&WaypointNode::callback_position, this, _1));
+
+    // Create a subscription for the "uav_mission_progress" topic
+    mission_progress_subscription =
+        this->create_subscription<interfaces::msg::MissionProgress>(
+            common_lib::topic_names::MissionProgress, 10,
+            std::bind(&WaypointNode::callback_mission_progress, this, _1));
+
+    // Create a publisher for the "uav_waypoint_command" topic
+    uav_waypoint_command_publisher =
+        this->create_publisher<interfaces::msg::UAVWaypointCommand>(
+            common_lib::topic_names::UAVWaypointCommand, 10);
+
+    // Initialize Event Loop
+    event_loop_timer = this->create_wall_timer(
+        std::chrono::milliseconds(event_loop_time_delta_ms),
+        std::bind(&WaypointNode::event_loop, this));
+}
+
+/**
+ * @brief Checks if the first loop was already triggered and returns the result.
+ *
+ * This function checks the state of the first loop and returns false if it was
+ * already triggered, otherwise it returns true. After returning the state, it
+ * sets the first loop state to false.
+ *
+ * @return False if the first loop was already triggered, true otherwise.
+ */
+bool WaypointNode::get_state_first_loop() {
```

Comment:
Why not make this constexpr?

Reply:
Not possible as this is not just a getter

Reply:
Constexpr is not limited to getters:
https://godbolt.org/z/v7qf86cWP

Reply:
That's true, but it doesn't compile if the function itself is defined in the cpp file:
```
error: inline function ‘constexpr bool WaypointNode::get_state_first_loop()’ used but never defined [-Werror]
   93 |     constexpr bool get_state_first_loop();
```

Reply:
Of cause inline(which includes constexpr) requires the definition and the declaration at the same point. No forward declarations are allowed in these cases. They wouldn´t make sense.

Reply:
I don't want complex functions (more complex than a simple getter) in my header.
-> Not gonna happen